valet/pathproc.go

```diff
@@ -120,6 +116,9 @@ func DoProcessFiles(paths <-chan FilePath, workPlan WorkPlan, maxThreads int) er
 		}
 		mu.Unlock()
 
+		wg.Add(1)
+		sem <- token{}
```

Comment:
What happens when the semaphore is full? Shouldn't the wg.Add(1) come afterwards?

Reply:
Yes, that's a more logical way to do it - increment when we're sure to be doing the work.